examples/all-clusters-app/all-clusters-common/src/rvc-modes.cpp

```diff
@@ -0,0 +1,167 @@
+
+// RVC Run
+
+static RvcRunModeDelegate * gRvcRunModeDelegate = nullptr;
+static ModeBase::Instance * gRvcRunModeInstance = nullptr;
+
+CHIP_ERROR RvcRunModeDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+void RvcRunModeDelegate::HandleChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
+{
+    uint8_t currentMode = mInstance->GetCurrentMode();
+
+    // Our business logic states that we can only switch into the mapping state from the idle state.
+    if (NewMode == RvcRunMode::ModeMapping && currentMode != RvcRunMode::ModeIdle)
+    {
+        response.status = to_underlying(ModeBase::StatusCode::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan::fromCharString("Change to the mapping mode is only allowed from idle"));
+    }
+
+    response.status = to_underlying(ModeBase::StatusCode::kSuccess);
+}
+
+CHIP_ERROR RvcRunModeDelegate::GetModeLabelByIndex(uint8_t modeIndex, chip::MutableCharSpan & label)
+{
+    if (modeIndex >= ArraySize(kModeOptions))
+    {
+        return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+    }
+    return chip::CopyCharSpanToMutableCharSpan(kModeOptions[modeIndex].label, label);
+}
+
+CHIP_ERROR RvcRunModeDelegate::GetModeValueByIndex(uint8_t modeIndex, uint8_t & value)
+{
+    if (modeIndex >= ArraySize(kModeOptions))
+    {
+        return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+    }
+    value = kModeOptions[modeIndex].mode;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR RvcRunModeDelegate::GetModeTagsByIndex(uint8_t modeIndex, List<ModeTagStructType> & tags)
+{
+    if (modeIndex >= ArraySize(kModeOptions))
+    {
+        return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+    }
+
+    if (tags.size() < kModeOptions[modeIndex].modeTags.size())
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    std::copy(kModeOptions[modeIndex].modeTags.begin(), kModeOptions[modeIndex].modeTags.end(), tags.begin());
+    tags.reduce_size(kModeOptions[modeIndex].modeTags.size());
+
+    return CHIP_NO_ERROR;
+}
+
+void emberAfRvcRunModeClusterInitCallback(chip::EndpointId endpointId)
+{
+    VerifyOrDie(endpointId == 1); // this cluster is only enabled for endpoint 1.
+    VerifyOrDie(gRvcRunModeDelegate == nullptr && gRvcRunModeInstance == nullptr);
+    gRvcRunModeDelegate = new RvcRunMode::RvcRunModeDelegate;
+    gRvcRunModeInstance =
+        new ModeBase::Instance(gRvcRunModeDelegate, 0x1, RvcRunMode::Id, chip::to_underlying(RvcRunMode::Feature::kOnOff));
+    gRvcRunModeInstance->Init();
+}
+
+// RVC Clean
+static RvcCleanModeDelegate * gRvcCleanModeDelegate = nullptr;
+static ModeBase::Instance * gRvcCleanModeInstance   = nullptr;
+
+CHIP_ERROR RvcCleanModeDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+void RvcCleanModeDelegate::HandleChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
+{
+    uint8_t rvcRunCurrentMode = mInstance->GetCurrentMode();
```

Comment:
@bzbarsky-apple This method of RVC Clean needs to look at an attribute of a different cluster (RVC Run). Perhaps I'm missing something, but the code above doesn't seem to be doing that.

Reply:
Ah!  I had totally misread what this was trying to do.  In that case, yes, we need to not use mInstance here!

Reply:
fixed using Petru's suggestion.